services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_core.py

```diff
@@ -327,30 +329,37 @@ async def _try_attach_pending_ec2s(
     )
 
 
-async def sorted_allowed_instance_types(app: FastAPI) -> list[EC2InstanceType]:
+async def _sorted_allowed_instance_types(app: FastAPI) -> list[EC2InstanceType]:
     app_settings: ApplicationSettings = app.state.settings
     assert app_settings.AUTOSCALING_EC2_INSTANCES  # nosec
     ec2_client = get_ec2_client(app)
 
     # some instances might be able to run several tasks
+    selected_names = list(
+        app_settings.AUTOSCALING_EC2_INSTANCES.EC2_INSTANCES_ALLOWED_TYPES
+    )
+    assert set(selected_names).issubset(typing.get_args(InstanceTypeType))  # nosec
+
     allowed_instance_types: list[
         EC2InstanceType
     ] = await ec2_client.get_ec2_instance_capabilities(
-        cast(
-            set[InstanceTypeType],
-            set(
-                app_settings.AUTOSCALING_EC2_INSTANCES.EC2_INSTANCES_ALLOWED_TYPES,
-            ),
-        )
+        cast(set[InstanceTypeType], set(selected_names))
     )
 
-    def _sort_according_to_allowed_types(instance_type: EC2InstanceType) -> int:
-        assert app_settings.AUTOSCALING_EC2_INSTANCES  # nosec
-        return list(
-            app_settings.AUTOSCALING_EC2_INSTANCES.EC2_INSTANCES_ALLOWED_TYPES
-        ).index(f"{instance_type.name}")
+    if selected_names:
+
+        def _as_selection(instance_type: EC2InstanceType) -> int:
+            assert app_settings.AUTOSCALING_EC2_INSTANCES  # nosec
+            return selected_names.index(f"{instance_type.name}")
 
-    allowed_instance_types.sort(key=_sort_according_to_allowed_types)
+        allowed_instance_types.sort(key=_as_selection)
+    else:
+        # NOTE An empty set to get_ec2_instance_capabilities it will return  ALL of the instances
```

Comment:
so here you are completely changing the behavior. And I don't think that is a good idea:
- Currently the `EC2_INSTANCES_ALLOWED_TYPES` variable defines what instance types are allowed. With this change now if that is empty any instance type will be allowed??? That is a complete change of behavior, and a loss of control. Also, that will probably break a load of things as normally that variable also contains what custom script is run on start, or which pre pulled images are used, etc etc..

Let's discuss that one, but for now I think this should be removed.

Reply:
I am not changing the behavior. It was already like that. Please read the description above.

Reply:
As agreed, i removed the warning and avoid empty sets